pkg/controller/master/nodedrain/nodedrain_controller.go

```diff
@@ -97,34 +97,56 @@ func (ndc *ControllerHandler) OnNodeChange(_ string, node *corev1.Node) (*corev1
 		if errors.Is(err, drainhelper.ErrNodeDrainNotPossible) {
 			logrus.WithFields(logrus.Fields{
 				"node_name": node.Name,
-			}).Warnf("Draining is not possible: %v", err)
+			}).Errorf("Enabling maintenance mode is impossible: %v", err)
 
-			nodeCopy := node.DeepCopy()
-
-			// Cleanup annotations to avoid recurring approaches to drain the node.
-			delete(nodeCopy.Annotations, ctlnode.MaintainStatusAnnotationKey)
-			delete(nodeCopy.Annotations, drainhelper.DrainAnnotation)
-			delete(nodeCopy.Annotations, drainhelper.ForcedDrain)
-
-			_, errUpdate := ndc.nodes.Update(nodeCopy)
+			_, errUpdate := ndc.cleanupMaintenanceModeAnnotations(node)
 			if errUpdate != nil {
-				return node, errors.Join(err, fmt.Errorf("failed to update node to clean up annotations: %w", errUpdate))
+				return node, errors.Join(err, errUpdate)
 			}
 
-			// Do not reconcile again.
+			// Abort ultimately; do not reconcile again.
 			return nil, err
 		}
 
 		return node, err
 	}
 
+	nonMigratableVMs, err := ndc.FindNonMigratableVMS(node)
+	if err != nil {
+		return node, fmt.Errorf("error getting non-migratable VMs: %w", err)
+	}
+
 	// List of VMs that need to be forcibly shutdown before maintenance
 	// mode.
 	shutdownVMs := make(map[string][]string)
 
 	if !forced {
 		var maintainModeStrategyVMs []string
 
+		// Make sure there are no VMs that are non-migratable. Abort the
+		// approach to place a node into maintenance mode if there are
+		// such VMs.
+		if len(nonMigratableVMs) > 0 {
+			reasons := make([]string, 0, len(nonMigratableVMs))
+			for condition, vms := range nonMigratableVMs {
+				reasons = append(reasons, fmt.Sprintf("%s cannot be migrated due to %s", strings.Join(vms, ","), condition))
+			}
+
+			msg := fmt.Sprintf("non-migratable VMs found: %s. Use the %q annotation to bypass this situation", strings.Join(reasons, "; "), drainhelper.ForcedDrain)
```

Comment:
```suggestion
			msg := fmt.Sprintf("non-migratable VMs found: %s. Use 'force drain' to perform a collective shutdown", strings.Join(reasons, "; "), drainhelper.ForcedDrain)
```
This can also be done via the UI.